accepted/better-obsoletion/better-obsoletion.md

```diff
@@ -0,0 +1,345 @@
+
+Trying to understand why, he clicks the warning ID `BCL0001` in the error list
+which takes him to a documentation page explaining the issues with
+`SecureString`. There he learns that this type doesn't work well for
+cross-platform scenarios due to the lack of automatic encryption facilities and
+the recommendation is to stop using it altogether. Bill knows that moving to the
+cloud also means they are considering moving to Linux, so he takes note that
+they also need to plan for removing their dependency on `SecureString`.
+
+## Requirements
+
+### Goals
+
+* **Global suppressions are specific**. Developers must be able to suppress
+  obsoletion diagnostics for all APIs that relate to the same technology, rather
+  than having to turn off all obsoletion diagnostics or having to resort to
+  suppressing all call sites individually.
+
+* **Documentation pages are easily accessible**. Developers must be able to go
+  from the diagnostic in the error list straight to the documentation that
+  provides the details.
+
+### Non-Goals
+
+* **Detecting specific usage patterns**. Sometimes, an API isn't obsoleted, just
+  a particular combination of arguments or specific values. If we need to
+  identify these, we'll author a custom analyzer.
+
+* **Being able to obsolete APIs in already shipped versions of .NET**. The
+  attribute must be applied to the API surface that the compiler sees and thus
+  must be present in the reference assemblies. We can only obsolete in new
+  versions of the platform. In other words, this doesn't support a sidecar file
+  that can be applied to an existing version of .NET.
+
+* **Being able to remove APIs in later versions**. See
+  [the discussion](#why-is-being-able-to-delete-apis-not-a-goal) in the Q&A
+  section.
+
+## Design
+
+The design proposal piggybacks on the existing `ObsoleteAttribute` that is
+already supported by all .NET compilers. The idea is to expand this a little to
+support the desired scenarios. Alternatives are listed afterwards.
+
+### API Changes
+
+We'd add the following properties to `ObsoleteAttribute`:
+
+```C#
+namespace System
+{
+    public sealed class ObsoleteAttribute : Attribute
+    {
+        // Existing:
+        //
+        // public ObsoleteAttribute();
+        // public ObsoleteAttribute(string message);
+        // public ObsoleteAttribute(string message, bool error);
+        // public bool IsError { get; }
+        // public string Message { get; }
+
+        // New:
+        public string DiagnosticId { get; set; }
+        public string UrlFormat { get; set; }
+        public string Url => UrlFormat == null ? null : string.Format(UrlFormat, DiagnosticId);
+    }
+}
+```
+
+Instead of taking the URL directly, the API takes a format string. This allows
+having a generic URL that includes the diagnostic ID. This avoids having to
+repeat the ID twice and thus making a copy & paste mistake.
+
+### Compiler Changes
+
```

Comment:
Maybe you are talking in more detail elsewhere, but doesn't the compiler now have two jobs:

- Recognize to ignore (or set to error) by DiagnosticId
- Create a message for the warning

Reply:
Clarified.